Description: blake3: Use the pure-Rust implementation
Author: Fabian Grünbichler <f.gruenbichler@proxmox.com>
Last-Update: 2025-08-05
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/vendor/blake3-1.8.2/Cargo.toml
+++ b/vendor/blake3-1.8.2/Cargo.toml
@@ -38,7 +38,7 @@
 ]
 
 [features]
-default = ["std"]
+default = ["std", "pure"]
 digest = ["dep:digest"]
 mmap = [
     "std",
--- a/vendor/blake3-1.8.2/build.rs
+++ b/vendor/blake3-1.8.2/build.rs
@@ -369,11 +369,13 @@
     println!("cargo:rerun-if-env-changed=CFLAGS");
 
     // Ditto for source files, though these shouldn't change as often.
-    for file in std::fs::read_dir("c")? {
-        println!(
-            "cargo:rerun-if-changed={}",
-            file?.path().to_str().expect("utf-8")
-        );
+    if !is_pure() {
+        for file in std::fs::read_dir("c")? {
+            println!(
+                "cargo:rerun-if-changed={}",
+                file?.path().to_str().expect("utf-8")
+            );
+        }
     }
 
     // When compiling with clang-cl for windows, it adds .asm files to the root
